src/shims/unix/fd.rs

```diff
@@ -383,9 +390,9 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
         // `File::read` never returns a value larger than `count`,
         // so this cannot fail.
         let result = file_descriptor
-            .read(communicate, &mut bytes, *this.tcx)?
+            .borrow_mut()
+            .read(communicate, &mut bytes, this)?
             .map(|c| i64::try_from(c).unwrap());
-        drop(file_descriptor);
```

Comment:
I think its still good to keep the FD for as short as possible, so please keep the `drop` around.

Reply:
That's right, fixed.